core/src/main/java/org/infinispan/xsite/events/XSiteViewMetrics.java

```diff
@@ -0,0 +1,85 @@
+package org.infinispan.xsite.events;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+import org.infinispan.commons.stat.MetricInfo;
+import org.infinispan.metrics.Constants;
+import org.infinispan.metrics.impl.MetricUtils;
+import org.infinispan.metrics.impl.MetricsRegistry;
+import org.infinispan.remoting.transport.Transport;
+
+class XSiteViewMetrics implements Constants {
+
+   private static final int OFFLINE = 0;
+   private static final int ONLINE = 1;
+   private static final int UNKNOWN = 2;
+   private static final Collection<MetricInfo> METRIC = List.of(MetricUtils.createGauge("SiteViewStatus", "Returns the site status in JGroups view (0 -> offline, 1 -> online, 2 -> n/a)", AtomicInteger::get, null));
+
+   private final Map<String, SiteStatusMetric> siteMetrics;
+   private final Supplier<MetricsRegistry> metricsRegistrySupplier;
+   private final Supplier<Transport> transportSupplier;
+   private SiteStatusMetric localSite;
+
+   XSiteViewMetrics(Supplier<MetricsRegistry> metricsRegistrySupplier, Supplier<Transport> transportSupplier) {
+      this.metricsRegistrySupplier = metricsRegistrySupplier;
+      this.transportSupplier = transportSupplier;
+      siteMetrics = new ConcurrentHashMap<>(4);
+   }
+
+   void start() {
+      var siteName = transportSupplier.get().localSiteName();
+      if (siteName == null) {
+         return;
+      }
+      localSite = create(transportSupplier.get().localSiteName());
+      localSite.status.set(ONLINE);
```

Comment:
Here we're setting the metrics to ONLINE for all of the nodes, not just the coordinator, consequently !coordinator nodes return 1 when they should be returning 2.

Reply:
The local site is always "online". You can't ever get an empty view :) 
Is it causing issues with the lambda?

Reply:
Argh, I accused the wrong part of the code for the issue I'm having, the clue should have been in the name :sweat_smile:

Currently all nodes are returning `1` for the remote-site, but I would expect the coordinator to return 1 and all other nodes return 2.

Reply:
Would it make sense to remove the local site name from the metrics? It is weird to have the local site name changing value since it should always be available.

Reply:
> Would it make sense to remove the local site name from the metrics?

+1, I don't think it adds any value as it should always be available.